capsules/src/net/ipv6/ipv6_send.rs

```diff
@@ -163,29 +164,56 @@ impl IP6SendStruct<'a> {
 
     // Returns EBUSY if the tx_buf is not there
     fn send_next_fragment(&self) -> ReturnCode {
-        self.ip6_packet
+        // Below code adds delay between fragments. Despite some efforts
+        // to fix this bug, I find that without it the receiving imix cannot
+        // receive more than 2 fragments in a single packet without hanging
+        // waiting for the third fragments.
+        let mut x = 0;
+        for i in 0..100000 {
```

Comment:
Can this at the very least become a timer callback, rather than spin loop? Adding a timeout delay before sending a fragment would not be crazy.

Reply:
I implemented this using a timer last night, but something about my changes triggered the #1160 bug to show up again, so I am unable to test it. I tried sprinkling some debug code around to see if that would hide the bug, but no luck..

Reply:
I went ahead and made some long overdue changes in addition to this change (cleaning up code related to cksum elision, fixing some bugs in checksum calculation, fixed some bugs when calculating the length of udp packets that have been compressed certain ways by 6lowpan, adding checksum verification on receive for udp/icmp). I made those changes, then rebased my delay fix on them, and the process fault bug went away, so I went ahead and pushed.